src/plots/linePlot.ts

```diff
@@ -22,6 +31,76 @@ export module Plots {
       this.attr("stroke-width", "2px");
     }
 
+    public x(): Plots.AccessorScaleBinding<X, number>;
+    public x(x: number | Accessor<number>): Line<X>;
+    public x(x: X | Accessor<X>, xScale: Scale<X, number>): Line<X>;
+    public x(x?: number | Accessor<number> | X | Accessor<X>, xScale?: Scale<X, number>): any {
+      if (xScale instanceof QuantitativeScale && this.autorangeMode() === "x") {
+        (<QuantitativeScale<X>>xScale).snapsDomain(!this._autorangeSmooth);
+      }
+
+      if (xScale == null) {
+        return super.x(<number | Accessor<number>>x);
+      } else {
+        return super.x(<X | Accessor<X>>x, xScale);
+      }
+
+    }
+
+    public y(): Plots.AccessorScaleBinding<number, number>;
+    public y(y: number | Accessor<number>): Line<X>;
+    public y(y: number | Accessor<number>, yScale: Scale<number, number>): Line<X>;
+    public y(y?: number | Accessor<number>, yScale?: Scale<number, number>): any {
+      if (yScale instanceof QuantitativeScale && this.autorangeMode() === "y") {
+        (<QuantitativeScale<number>>yScale).snapsDomain(!this._autorangeSmooth);
+      }
+      return super.y(y, yScale);
+    }
+
+    public autorangeMode(): string;
+    public autorangeMode(autorangeMode: string): Line<X>;
+    public autorangeMode(autorangeMode?: string): any {
+      if (this.autorangeSmooth() && autorangeMode === "x" || autorangeMode === "y") {
+        if (autorangeMode === "x" && this.x() && this.x().scale && this.x().scale instanceof QuantitativeScale) {
+          (<QuantitativeScale<X>>this.x().scale).snapsDomain(!this.autorangeSmooth());
+        }
+
+        if (autorangeMode === "y" && this.y() && this.y().scale && this.y().scale instanceof QuantitativeScale) {
+          (<QuantitativeScale<number>>this.y().scale).snapsDomain(!this.autorangeSmooth());
+        }
+      }
+      return super.autorangeMode(autorangeMode);
+    }
+
+    /**
+     * Gets whether or not the autoranging is done smoothly.
+     */
+    public autorangeSmooth(): boolean;
+    /**
+     * Sets whether or not the autorange is done smoothly.
+     *
+     * Smooth autoranging is done by making sure lines always exit on the left / right side of the plot
+     * and deactivating the nice domain feature on the scales
+     */
+    public autorangeSmooth(autorangeSmooth: boolean): Plots.Line<X>;
+    public autorangeSmooth(autorangeSmooth?: boolean): any {
+      if (autorangeSmooth == null) {
+        return this._autorangeSmooth;
+      }
+      this._autorangeSmooth = autorangeSmooth;
+
+      if (this.autorangeMode() === "x" && this.x() && this.x().scale && this.x().scale instanceof QuantitativeScale) {
+        (<QuantitativeScale<X>>this.x().scale).snapsDomain(!autorangeSmooth);
+      }
+
+      if (this.autorangeMode() === "y" && this.y() && this.y().scale && this.y().scale instanceof QuantitativeScale) {
```

Comment:
The problem is that they are slightly different. Not sure if it is worth extracting it